internal/pkg/skuba/addons/kured_test.go

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2019 SUSE LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package addons
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/SUSE/skuba/internal/pkg/skuba/kubernetes"
+	"k8s.io/apimachinery/pkg/util/version"
+)
+
+func TestGetKuredImage(t *testing.T) {
+	type args struct {
+		imageTag string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		// TODO: Add test cases.
+		{
+			name: "TestGetKuredImage",
+			args: args{
+				imageTag: "1.2.0-rev4",
+			},
+			want: "registry.suse.de/devel/caasp/4.0/containers/containers/caasp/v4/kured:1.2.0-rev4",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := GetKuredImage(tt.args.imageTag); got != tt.want {
+				t.Errorf("GetKuredImage() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_renderContext_KuredImage(t *testing.T) {
+	type fields struct {
+		addon  Addon
+		config AddonConfiguration
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   string
+	}{
+		// TODO: Add test cases.
+		{
+			name: "Test_renderContext_KuredImage_1.15.2",
+			fields: fields{
+				addon: Addon{
+					addon:             kubernetes.Kured,
+					templater:         renderKuredTemplate,
+					callbacks:         nil,
+					addonPriority:     normalPriority,
+					getImageCallbacks: []getImageCallback{GetKuredImage},
+				},
+				config: AddonConfiguration{
+					ClusterVersion: version.MustParseSemantic("1.15.2"),
+					ControlPlane:   "",
+					ClusterName:    "",
+				},
+			},
+			want: "registry.suse.de/devel/caasp/4.0/containers/containers/caasp/v4/kured:[[:digit:]]{1,}.[[:digit:]]{1,}.[[:digit:]]{1,}[-rev[:digit:]{1,}]{0,1}",
+		},
+		{
+			name: "Test_renderContext_KuredImage_1.16.2",
+			fields: fields{
+				addon: Addon{
+					addon:             kubernetes.Kured,
+					templater:         renderKuredTemplate,
+					callbacks:         nil,
+					addonPriority:     normalPriority,
+					getImageCallbacks: []getImageCallback{GetKuredImage},
+				},
+				config: AddonConfiguration{
+					ClusterVersion: version.MustParseSemantic("1.16.2"),
+					ControlPlane:   "",
+					ClusterName:    "",
+				},
+			},
+			want: "registry.suse.de/devel/caasp/4.0/containers/containers/caasp/v4/kured:[[:digit:]]{1,}.[[:digit:]]{1,}.[[:digit:]]{1,}[-rev[:digit:]{1,}]{0,1}",
```

Comment:
I am not sure is it better to leverage existed const variable [ImageRepository](https://github.com/SUSE/skuba/blob/d204cff5631e237a38f6e305225d7bc3a23bad1f/pkg/skuba/development_constants.go#L24)?

Reply:
agreed. done.